openssl-dynamic/src/main/java/org/apache/tomcat/jni/socket/AprSocketContext.java

```diff
@@ -797,10 +797,16 @@ void prepare() throws IOException {
         }
 
         void unblock() {
-            try (java.net.Socket sock = new java.net.Socket()) {
-                // Easiest ( maybe safest ) way to interrupt accept
-                // we could have it in non-blocking mode, etc
-                sock.connect(new InetSocketAddress("127.0.0.1", port));
+            java.net.Socket sock = new java.net.Socket();
+
+            try {
+                try {
```

Comment:
@Scottmitch 

>  Lets move the finally to the outer try and remove the inner try.

The code is correct as-is.  Even if I were to do what you say here, you would still need another try/finally block because ``close`` throws ``IOException``.

Considering every exception is ignored here, I would argue that this style makes slightly more sense.  That being said, I don't know if there is a util class that supports closing and swallowing the exception.  If that's available, this would be much nicer.